indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamEndSequenceNumbers.java

```diff
@@ -147,6 +148,38 @@ public SeekableStreamSequenceNumbers<PartitionIdType, SequenceOffsetType> plus(
     }
   }
 
+  @Override
+  public int compareTo(SeekableStreamSequenceNumbers<PartitionIdType, SequenceOffsetType> other, Comparator<SequenceOffsetType> comparator)
+  {
+    if (this.getClass() != other.getClass()) {
+      throw new IAE(
+          "Expected instance of %s, got %s",
+          this.getClass().getName(),
+          other.getClass().getName()
+      );
+    }
+
+    final SeekableStreamEndSequenceNumbers<PartitionIdType, SequenceOffsetType> otherStart =
+        (SeekableStreamEndSequenceNumbers<PartitionIdType, SequenceOffsetType>) other;
+
+    if (stream.equals(otherStart.stream)) {
+      //Same stream, compare the offset
+      boolean res = false;
+      for (Map.Entry<PartitionIdType, SequenceOffsetType> entry : partitionSequenceNumberMap.entrySet()) {
+        PartitionIdType partitionId = entry.getKey();
+        SequenceOffsetType sequenceOffset = entry.getValue();
+        if (otherStart.partitionSequenceNumberMap.get(partitionId) != null && comparator.compare(sequenceOffset, otherStart.partitionSequenceNumberMap.get(partitionId)) > 0) {
```

Comment:
If otherStart for partitionId is null and sequenceOffset is not null, should the comparator return 1?

Reply:
@AmatyaAvadhanula, I think this will happen, when the partition reporting offset for the first time, right? So, for this case, there will be no need to retry, As startMetadataMatchesExisting will be true. And it will update the datastore.

Reply:
Would startMetadataMatchesExisting be true if the entire metadata is null or if any one of the partitions has it as null as well? 
I think it's just the former

Reply:
It will be true for both. For First the metadata from DB is null and it checks for validStart(). And for the second it will try to match old and new metadata, using a+b equals b+a. So, it will also be true.